source/agora/utils/Test.d

```diff
@@ -627,106 +620,413 @@ unittest
+        }
+
+        // Return the result and reset this
+        this.inputs = null;
+        this.leftover = Output.init;
+        scope (success) this.data = Transaction.init;
+        return this.data;
+    }
+
+    /***************************************************************************
+
+        Finds a well-known `KeyPair` corresponding to the `PublicKey`.
+
+        Params:
+            address = finding `PublicKey`
+
+        Returns:
+            returns KeyPair.init if `address` is not found.
+
+    ***************************************************************************/
+
+    private KeyPair getOwnerKeyPair (scope const PublicKey address) @safe
+    {
+        // FIXME: Using the binary version didn't work (?)
+        if (address.toString() == "GCOQEOHAUFYUAC6G22FJ3GZRNLGVCCLESEJ2AXBIJ5BJNUVTAERPLRIJ")
+            return  () @trusted {
+                return KeyPair.fromSeed(Seed.fromString(
+                    "SCT4KKJNYLTQO4TVDPVJQZEONTVVW66YLRWAINWI3FZDY7U4JS4JJEI4"));
+            }();
+
+        return WK.Keys[address];
+    }
+
+    /***************************************************************************
+
+        Resets the state and changes the address of the refund transaction
+
+        Ideally one should provide the correct address for the refund
+        transaction in the constructor, however if for some reason it is not
+        known in advance, or if the `Output` or `Transaction` overload is used,
+        this function provides a convenient mean to change the refund address.
+
+        Params:
+            toward = Refund address to use for the transaction being built
+
+        Returns:
+            Reference to `this` for easy chaining
+
+    ***************************************************************************/
+
+    public ref typeof(this) refund (scope const PublicKey toward)
+        @safe return
+    {
+        assert(this.inputs.length > 0);
+
+        this.leftover = Output(Amount(0), toward);
+        this.inputs.each!(val => this.leftover.value.mustAdd(val.output.value));
+        this.data.outputs = null;
+
+        return this;
+    }
+
+    /***************************************************************************
+
+        Splits the attached input into multiple outputs of the given amounts.
+
+        Any leftover will remain in the refund transaction.
+        The value drawn (`amount * toward.length`) must be lesser or equal
+        to the fund currently attached and in the refund transaction.
+
+        Params:
+            KeyRange = A range of `PublicKey`
+            amount = `Amount` to distribute to each `PublicKey`
+            toward = Array of `PublicKey` to give `amount` to
+
+        Returns:
+            Reference to `this` for easy chaining
+
+    ***************************************************************************/
+
+    public ref typeof(this) draw (KeyRange) (Amount amount, scope KeyRange toward)
```

Comment:
Is `draw` supposed to be short for `withdraw`? It's ok to keep the name, I just wonder.

Reply:
Indeed. I think it's proper English though ? [Source](https://dictionary.reverso.net/english-cobuild/to+draw+money+from+an+account). CC @bpalaggi

Reply:
I think this is not divided into fixed values, but filled with the user's chosen values.

Reply:
> Indeed. I think it's proper English though ? [Source](https://dictionary.reverso.net/english-cobuild/to+draw+money+from+an+account). CC @bpalaggi

Yes, that works.

Reply:
Right maybe this should be named `distribute` or just `dist` for short?

Reply:
But if it's an already existing terminology, that's ok to keep it this way.

Reply:
Okay let's leave it at `draw` for the time being